src/shared/logger.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import * as winston from 'winston'
+
+const localize = nls.loadMessageBundle()
+let logSettings: LogSettings
+
+class LogSettings {
+    private readonly outputChannel: vscode.OutputChannel
+    private readonly logger: winston.Logger
+    private readonly logPath: string
+
+    public constructor(outputChannel: vscode.OutputChannel, logPath: string, level: string = 'info') {
+        this.outputChannel = outputChannel
+        this.logger = winston.createLogger({
+            transports: [new winston.transports.File({ filename: logPath })]
+        })
+        this.logPath = logPath
+        // TODO: check if a default log level already exists--we can add a field for this to the settings page.
+        this.setLevel(level)
+    }
+
+    public getOutputChannel(): vscode.OutputChannel {
+        return this.outputChannel
+    }
+
+    public getLogger(): winston.Logger {
+        return this.logger
+    }
+
+    public setLevel(level: string): void {
```

Comment:
Instead of validating the level at runtime, change it's type to `'info' | 'warn' | 'error'` and the compiler will validate it for us.

Reply:
Could alternately use the `winston` preset levels as well.

Reply:
[String enum](https://mariusschulz.com/blog/typescript-2-4-string-enums) please :)